tensorboard/webapp/notification_center/_redux/notification_center_effects.ts

```diff
@@ -12,3 +12,67 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the License for the specific language governing permissions and
 limitations under the License.
 ==============================================================================*/
+import {Injectable} from '@angular/core';
+import {Actions, createEffect, ofType, OnInitEffects} from '@ngrx/effects';
+import {Action, createAction, Store} from '@ngrx/store';
+import {Observable, of} from 'rxjs';
+import {catchError, map, mergeMap} from 'rxjs/operators';
+import {State} from '../../app_state';
+import {NotificationCenterDataSource} from '../_data_source/index';
+import {Notification} from './notification_center_types';
+import * as actions from './notification_center_actions';
+
+/** @typehack */ import * as _typeHackNgrxEffects from '@ngrx/effects';
+/** @typehack */ import * as _typeHackModels from '@ngrx/store/src/models';
+/** @typehack */ import * as _typeHackStore from '@ngrx/store';
+
+export const initAction = createAction('[NotificationCenter Effects] Init');
+
+@Injectable()
+export class NotificationCenterEffects implements OnInitEffects {
+  constructor(
+    private readonly actions$: Actions,
+    private readonly store: Store<State>,
+    private readonly dataSource: NotificationCenterDataSource
+  ) {}
+
+  /** @export */
+  ngrxOnInitEffects(): Action {
+    return initAction();
+  }
+
+  /**
+   * Initiates notifications fetching.
+   *
+   * @export
+   */
+  initialNotificationFetch$ = createEffect(
+    () => {
+      return this.actions$.pipe(
+        ofType(initAction),
+        mergeMap(() => this.fetchNotification())
+      );
+    },
+    {dispatch: false}
+  );
+
+  private fetchNotification() {
```

Comment:
specify the return type. Specifically, you probably don't need L70.

Reply:
specify the return type as `Observable<void>`
and replace L70 with `return of();`

w/o return in L70 results in type unmatch error:
```
Argument of type '() => void' is not assignable to parameter of type '(err: any, caught: Observable<void>) => ObservableInput<any>'.
  Type 'void' is not assignable to type 'ObservableInput<any>'.
```

would love to hear any suggestions in addition to example in [metrics/](https://cs.opensource.google/tensorflow/tensorboard/+/master:tensorboard/webapp/metrics/effects/index.ts;l=183?q=dataSource%20catchError%20-f:test&ss=tensorflow%2Ftensorboard:tensorboard%2Fwebapp%2F)